aptos-move/block-executor/src/view.rs

```diff
@@ -353,7 +432,82 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TAggregator
         state_key: &Self::IdentifierV1,
         _mode: AggregatorReadMode,
     ) -> anyhow::Result<Option<StateValue>> {
-        // TODO: Integrate aggregators.
+        // TODO: Integrate aggregators V1. That is, we can lift the u128 value
+        //       from the state item by passing the right layout here. This can
+        //       be useful for cross-testing the old and the new flows.
+        // self.get_resource_state_value(state_key, Some(&MoveTypeLayout::U128))
         self.get_resource_state_value(state_key, None)
     }
+
+    fn generate_aggregator_v2_id(&self) -> Self::IdentifierV2 {
+        match &self.latest_view {
+            ViewState::Sync(state) => (state.counter.fetch_add(1, Ordering::SeqCst) as u64).into(),
+            ViewState::Unsync(state) => {
+                let mut counter = state.counter.borrow_mut();
+                let id = (*counter as u64).into();
+                *counter += 1;
+                id
+            },
+        }
+    }
+}
+
+// For aggregators V2, values are replaced with identifiers at deserialization time,
+// and are replaced back when the value is serialized. The "lifted" values are cached
+// by the `LatestView` in the aggregators multi-version data structure.
+impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> ValueExchange
+    for LatestView<'a, T, S, X>
+{
+    fn try_exchange(
+        &self,
+        layout: &MoveTypeLayout,
+        value_to_exchange: Value,
+    ) -> TransformationResult<Value> {
+        let identifier = self.generate_aggregator_v2_id();
+        match &self.latest_view {
+            ViewState::Sync(state) => {
+                // TODO(aggregator): Integrate other types as well.
+                let base_value = value_to_exchange
```

Comment:
This was my understanding from the name "try_exchange". I thought `value_to_exchange` could be any resource, and this resource could contain many aggregators. We need to exchange each of the aggregators in the resource. 
But it seems like `value_to_exchange` could only be a `u128`?

Reply:
my understanding is that this is like a visitor pattern - try_exchange is called on every instance of aggregator - i.e. if resource has 3 aggregators, this function will be called 3 times.

Reply:
Yes, we have a visitor that does serialization/deserialization, and exchange is just added on top. SO it replaces the current node which is visited.